ipaserver/install/bindinstance.py

```diff
@@ -423,17 +423,15 @@ def zonemgr_callback(option, opt_str, value, parser):
             encoding = getattr(sys.stdin, 'encoding', None)
             if encoding is None:
                 encoding = 'utf-8'
-            value = value.decode(encoding)
+            if not isinstance(value, unicode):
+                value = value.decode(encoding)
+
             validate_zonemgr_str(value)
         except ValueError as e:
             # FIXME we can do this in better way
             # https://fedorahosted.org/freeipa/ticket/4804
             # decode to proper stderr encoding
-            stderr_encoding = getattr(sys.stderr, 'encoding', None)
-            if stderr_encoding is None:
-                stderr_encoding = 'utf-8'
-            error = unicode(e).encode(stderr_encoding)
-            parser.error("invalid zonemgr: " + error)
+            parser.error("invalid zonemgr: " + unicode(e))
```

Comment:
I suppose the original purpose of this encoding excercise was to do
`parser.error(b"invalid zonemgr: " + error)`